src/main/java/pl/edu/mimuw/queue/AbstractIntQueue.java

```diff
@@ -2,38 +2,71 @@
 
 public abstract class AbstractIntQueue {
 
-  // TODO: you can make changes with this class fields, constructors
-  //  but also add some methods but the specified methods cannot be changed (you
-  //  can change them not to be abstract and provide some implementation for them,
-  //  but they have to have the same names, arguments and returned values)
+  /**
+   * front to zawsze reprezentacja elementu, ktory zostanie pierwszy zdjety z kolejki lub null gdy jest pusta
+   */
+  protected IntQueueNode front;
+  protected int size;
 
   /**
    * Adds element to the queue.
-   *
    * @throws NullPointerException if the specified element is null
    */
-  public abstract void offer(Integer x);
+  public abstract void offer(Integer x);  
 
   /**
    * @return the head of this queue, or {@code null} if this queue is empty
    * and don't remove the element from the queue
    */
-  public abstract Integer peek();
+  public Integer peek() {
+    if (front == null) {
+      return null;
+    }
+    else
+    {
+      return front.getValue();
+    }
+  }
 
   /**
    * @return the head of this queue, or {@code null} if this queue is empty
    * and remove the element from the queue
    */
-  public abstract Integer poll();
-
+  public Integer poll(){
+    if (front == null) {
+      return null;
+    }
+    else
+    {
+      IntQueueNode returned = front;
+      front = front.getNext();
+      size--;
+      return returned.getValue();
+    }
+  }
   /**
    * @return readable representation of ordered queue elements
    */
   @Override
-  public abstract String toString();
+  public String toString(){
+    StringBuilder sb = new StringBuilder();
+    IntQueueNode currentNode = front;
+    sb.append("Queue contains: ");
+    if(front == null){
+      sb.append("NOTHING");
+      return sb.toString();
+    }
+    while(currentNode != null){
+      sb.append(currentNode.getValue()+" ");
```

Comment:
jak używamy `StringBuildera` to już staramy sie nie robić `+` na stringach - lepiej 2 razy `append`